### PR TITLE
language level updates

### PR DESCRIPTION
**Pull request by FromTheCosmos**
rewrote some methods up to 1.8 language level which introduces things like multi-catch support and collapsing of iterating while loops into single line for loops as well as removed redundant array creations for calling varargs

### CODE REVIEW

**Comment by FromTheCosmos** [comment]
rewrote some methods up to 1.8 language level which introduces things like multi-catch support and collapsing of iterating while loops into single line for loops as well as removed redundant array creations for  calling varargs

File: src/main/java/anon/demonx/DemonX.java [hunk 1]
```diff
@@ -37,164 +37,143 @@
 import javax.management.remote.JMXConnector;
 import javax.management.remote.JMXConnectorFactory;
 import javax.management.remote.JMXServiceURL;
+
```
Comment (hehnope) [inline src/main/java/anon/demonx/DemonX.java]: Remove the space

File: src/main/java/anon/demonx/DemonX.java [hunk 2]
```diff
@@ -37,164 +37,143 @@
...
 import org.json.simple.JSONObject;
 import org.apache.commons.cli.*;
+
```
Comment (hehnope) [inline src/main/java/anon/demonx/DemonX.java]: Remove the space

File: src/main/java/anon/demonx/DemonX.java [hunk 3]
```diff
@@ -37,164 +37,143 @@
...
-      // nothing, but easier on the eyes...
-    } catch (IOException localIOException) {
-      // nothing, but easier on the eyes...
-    }
 
-    try {
-      mbai = mbi.getAttributes();
-      MBeanAttributeInfo[] arrayOfMBeanAttributeInfo1;
-      int j = (arrayOfMBeanAttributeInfo1 = mbai).length;
+    private static JSONObject ja;
```
Comment (hehnope) [inline src/main/java/anon/demonx/DemonX.java]: You have a space before this, remove it.

File: src/main/java/anon/demonx/DemonX.java [hunk 4]
```diff
@@ -37,164 +37,143 @@
...
-  public static void main(String[] args)
-    throws Exception
-  {
-    LogManager.getLogManager().reset();
+    public static void main(String[] args) throws IOException {
 
-    String host = null;
-    int port = 0;
-    ja = new JSONObject();
+        LogManager.getLogManager().reset();
```
Comment (hehnope) [inline src/main/java/anon/demonx/DemonX.java]: There is a space before this, remove it

File: src/main/java/anon/demonx/DemonX.java [hunk 5]
```diff
@@ -37,164 +37,143 @@
...
-
-    // Start connection to JMX
-    JMXServiceURL url = new JMXServiceURL(String.format("service:jmx:rmi:///jndi/rmi://%s:%d/jmxrmi", new Object[] { jmxHost, jmxPort }));
-    JMXConnector jmxc = null;
-    try {
-      jmxc = JMXConnectorFactory.connect(url, environment);
-    } catch (IOException e) {
-      System.out.println(e);
-      System.exit(1);
-    } catch (Exception e) {
```
Comment (hehnope) [inline src/main/java/anon/demonx/DemonX.java]: Why did you remove the catch all exception?
